### PR TITLE
new: support for icmp type/code notation

### PR DESCRIPTION
**Pull request by t00f**
This PR adds the type/code notation for icmp.

> Ref https://github.com/aporeto-inc/aporeto-devs/issues/704

### CODE REVIEW

File: custom_validations.go [hunk 1]
```diff
@@ -674,12 +646,89 @@ func ValidateServicePort(attribute string, servicePort string) error {
...
+
+// isNumberBetween check if a string is a number within the min and max boundaries.
+func isNumberBetween(s string, min int, max int) (int, error) {
+
+	i, err := strconv.Atoi(s)
+	if err != nil {
+		return -1, fmt.Errorf("'%s' is not a valid number", s)
+	}
+
+	if i < min || i > 255 {
```
Comment (ericrpowers) [inline custom_validations.go:727]: Shouldn't `max` be used here?
```
if i < min || i > max {
```

Comment (t00f) [reply custom_validations.go:727]: You are definitely right :)

File: custom_validations.go [hunk 2]
```diff
@@ -674,12 +646,89 @@ func ValidateServicePort(attribute string, servicePort string) error {
...
+
+	if typeCode == "" {
+		return makeValidationError(attribute, fmt.Sprintf("protocol '%s' has missing type/code information", protocol))
+	}
+
+	parts := strings.SplitN(typeCode, "/", 2)
+
+	// Validate type
+	if _, err := isNumberBetween(parts[0], 0, 255); err != nil {
+		return makeValidationError(attribute, fmt.Sprintf("protocol '%s' has invalid type notation. %s", protocol, err.Error()))
```
Comment (ericrpowers) [inline custom_validations.go:676]: `err.Error()` can just be `err`

File: custom_validations.go [hunk 3]
```diff
@@ -674,12 +646,89 @@ func ValidateServicePort(attribute string, servicePort string) error {
...
+
+			// Validate code range "x:y" with x < y.
+			if strings.Contains(code, ":") {
+
+				rangeParts := strings.SplitN(code, ":", 2)
+
+				// Validate left part
+				codeLeft, err := isNumberBetween(rangeParts[0], 0, 255)
+				if err != nil {
+					return makeValidationError(attribute, fmt.Sprintf("protocol '%s' has invalid code notation. %s", protocol, err.Error()))
```
Comment (ericrpowers) [inline custom_validations.go:693]: `err.Error()` can just be `err`

File: custom_validations.go [hunk 4]
```diff
@@ -674,12 +646,89 @@ func ValidateServicePort(attribute string, servicePort string) error {
...
+				}
+
+				// Validate right part
+				codeRight, err := isNumberBetween(rangeParts[1], 0, 255)
+				if err != nil {
+					return makeValidationError(attribute, fmt.Sprintf("protocol '%s' has invalid code notation. %s", protocol, err.Error()))
```
Comment (ericrpowers) [inline custom_validations.go:699]: `err.Error()` can just be `err`

File: custom_validations.go [hunk 5]
```diff
@@ -674,12 +646,89 @@ func ValidateServicePort(attribute string, servicePort string) error {
...
+
+				// Validate order
+				if codeLeft >= codeRight {
+					return makeValidationError(attribute, fmt.Sprintf("protocol '%s' has invalid code range order", protocol))
+				}
+
+			} else {
+				// Validate unique code
+				if _, err := isNumberBetween(code, 0, 255); err != nil {
+					return makeValidationError(attribute, fmt.Sprintf("protocol '%s' has invalid code notation: %s", protocol, err.Error()))
```
Comment (ericrpowers) [inline custom_validations.go:710]: `err.Error()` can just be `err`

File: custom_validations_test.go [hunk 6]
```diff
@@ -324,6 +325,133 @@ func TestValidateServicePorts(t *testing.T) {
...
+			},
+			true,
+		},
+		{
+			"icmp/256 is not valid",
+			args{
+				[]string{"icmp/256"},
+			},
+			true,
+		},
```
Comment (ericrpowers) [inline custom_validations_test.go:384]: Probably meant for second one to be `icmp6/256`

Comment (t00f) [reply custom_validations_test.go:384]: It looks like a duplicate

**Pull request merged**